### PR TITLE
Refactored code with proper modular design

### PR DESCRIPTION
**Pull request by KapilTR**
Following scenarios to review
- converted helper.js class methods as static
- removed all underscores from function naming
- indentation and design fix
- replaced all double tabs with single tabs
- removed promises and break down methods of setup class into small modular methods.

### CODE REVIEW

**Comment by jvtezsure** [comment]
Give proper title and description of change for the pull reuest so it's easier for reviwer to get the idea.
Similarly each commit should describe what's in the commit.

File: modules/accounts/accounts.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+constructor(){  
+    this.helper = new Helper();
+}
+
+confFile = __dirname + '/../../config.json';
+jsonfile = require('jsonfile');
+config = jsonfile.readFileSync(confFile);
+
+async _setProvider(){
```
Comment (jvtezsure) [inline modules/accounts/accounts.js]: Why do we need underscore for variable names?

File: modules/accounts/accounts.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    if (config.provider) {
+        eztz.node.setProvider(config.provider);
+    }  
+    const _sodium = require('libsodium-wrappers');
+    await _sodium.ready;
+    eztz.library.sodium = _sodium;
+}
+
+restoreAlphanetAccount(accountLabel, accountFilePath) {
+    const fs = require("fs");
```
Comment (jvtezsure) [inline modules/accounts/accounts.js]: use single quotes everywhere.

File: modules/accounts/accounts.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+      this.addIdentity(accountLabel, alphakeys.sk, alphakeys.pk, alphakeys.pkh, accountJSON.secret);
+      this.addAccount('aplha_'+ accountLabel, alphakeys.pkh, accountLabel);
+      return this.helper.output(`successfully restored testnet faucet account: ${accountLabel}-${alphakeys.pkh}`);
+    } catch(error) {
+      return this.helper.outputError(` occured while restroing account : ${error}`);
+    }
+}
+
+async activateAlphanetAccount(account) {
+const keys = this.getKeys(account);
```
Comment (jvtezsure) [inline modules/accounts/accounts.js]: Fix indentation of this method.

File: modules/contract/contracts.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+var eztz = {};
+var config = jsonfile.readFileSync(confFile);
+const ConseilJS = '../../lib/conseiljs';
+
+const {Helper} = require('../../helper');
+
+class Contracts{
+
+constructor(){  
+    this.helper = new Helper();
```
Comment (jvtezsure) [inline modules/contract/contracts.js]: Helper module has static functions. You don't need to create an instance of that. You can call the functions directly.

File: modules/accounts/accounts.js [hunk 5]
```diff
@@ -0,0 +1,239 @@
+const confFile = __dirname + '/../../config.json';
+const jsonfile = require('jsonfile');
+var eztz = {};
+var config = jsonfile.readFileSync(confFile);
+
+const {Helper} = require('../../helper');
+
+class Accounts{
+
+constructor(){
```
Comment (jvtezsure) [inline modules/accounts/accounts.js]: Fix indentation for whole file.

File: modules/setup/setup.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
+const childprocess = require("child_process");
+const {Helper} = require('../../helper');
+
+const Docker = require("dockerode");
+var docker = new Docker({ socketPath: "/var/run/docker.sock" });
+const imageTag = "tezsureinc/tezster:1.0.0";
```
Comment (jvtezsure) [inline modules/setup/setup.js]: For constant variable, use all CAPITAL letters with underscore for any seperation such as IMAGE_TAG.

File: helper.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+const cliColors = {
+    red : '31m',
+    yellow : '33m',
+    cyan : '36m',
+    white : '37m',
+    green : '32m',
+};
+
+class Helper{
```
Comment (jvtezsure) [inline helper.js]: make this class work as a static module. We should be able to use it without creating an instance of it.

File: tezster-manager.js [hunk 8]
```diff
@@ -1,416 +1,85 @@
...
-
-function getProvider(){    
-    if (config.provider){
-        return outputInfo(config.provider);
-    }else{
-        return outputInfo("No provider is set");
-    } 
-}
+constructor(){
+    this.Accounts = new Accounts();
```
Comment (jvtezsure) [inline tezster-manager.js]: private varialbes start with small case.

File: tezster-manager.js [hunk 9]
```diff
@@ -1,416 +1,85 @@
...
+    this.Transactions = new Transactions();
+} 
 
-function setProvider(args){    
-    config.provider = args[0];
-    jsonfile.writeFile(confFile, config);
-    eztz.node.setProvider(config.provider);
-    return outputInfo("Provider updated to " + config.provider);
+setup(){
+    this.Setup._setup();
```
Comment (jvtezsure) [inline tezster-manager.js]: function name should not have underscore.

File: helper.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+const cliColors = {
+    red : '31m',
+    yellow : '33m',
```
Comment (jvtezsure) [inline helper.js]: you can create a utils folder for helper.js file 
It shouldn't be at the same level as tezster.js

**Comment by jvtezsure** [comment]
Overall code is well structured. Please fix the indentation issues and other comments.

**Comment by jvtezsure** [comment]
Change the PR title to some meaningful header.

File: modules/accounts/accounts.js [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+            keys = Helper.findKeyObj(config.identities, f.identity);
+        } else if (f = Helper.findKeyObj(config.contracts, account)) {
+            keys = Helper.findKeyObj(config.identities, f.identity);
+        }
+        return keys;
+    }
+
+    addIdentity(label, sk, pk, pkh, secret) {
+        config.identities.push({
+        sk : sk,
```
Comment (jvtezsure) [inline modules/accounts/accounts.js]: Fix indentation.

File: modules/accounts/accounts.js [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+        pkh : pkh,
+        label : label,
+        secret: secret || ''
+        });
+        jsonfile.writeFile(confFile, config);
+    }
+
+    addAccount(label, pkh, identity) {
+        config.accounts.push({
+        label : label,
```
Comment (jvtezsure) [inline modules/accounts/accounts.js]: indentation.

File: modules/accounts/accounts.js [hunk 13]
```diff
@@ -0,0 +1,233 @@
+const confFile = __dirname + '/../../config.json';
+const jsonfile = require('jsonfile');
+var eztz = {};
+var config = jsonfile.readFileSync(confFile);
+
+const { Helper } = require('../../helper');
+
+class Accounts{
+
+    confFile = __dirname + '/../../config.json';
```
Comment (jvtezsure) [inline modules/accounts/accounts.js]: Why are you reimporting the same variables ( confFile, jsonfile, config) again. You can create a separate module to handle reading and writing to the file or at least handling the common functionality from accounts.js, transactions.js. That way code becomes easier to maintain.

File: modules/contract/contracts.js [hunk 14]
```diff
@@ -0,0 +1,246 @@
+const confFile = __dirname + '/../../config.json';
+const jsonfile = require('jsonfile');
+var eztz = {};
+var config = jsonfile.readFileSync(confFile);
+const ConseilJS = '../../lib/conseiljs';
```
Comment (jvtezsure) [inline modules/contract/contracts.js:5]: You're not using this variable in this file.

File: modules/contract/contracts.js [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+      }
+      await this.loadTezsterConfig(); 
+      
+      let result = await this.invokeContract(args[0], args[1], args[2]);
+      console.log(result);
+      console.log(Helper.outputInfo(`If you're using babylonnet node, use https://babylonnet.tzstats.com to check contract/transactions`));
+  }
+
+  async getStorage() {
+      var args = process.argv.slice(3);
```
Comment (jvtezsure) [inline modules/contract/contracts.js]: You need to pass this arguments to this function. You shouldn't access arguments directly like this here. 
Please fix this in other places as well.

File: package.json [hunk 16]
```diff
@@ -40,6 +40,8 @@
     "node-cmd": "^3.0.0",
     "node-fetch": "^2.6.0",
     "pbkdf2": "^3.0.17",
+    "request": "^2.81.0",
+    "tezster2": "^0.7.5",
```
Comment (jvtezsure) [inline package.json]: What is tezster2 package?

File: package.json [hunk 17]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tezster",
-  "version": "0.2.0",
+  "version": "2.0.0",
```
Comment (jvtezsure) [inline package.json]: Why are you bumping the version?

File: helper.js [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+        return s + (j ? i.substr(0, j) + t : "") + i.substr(j).replace(/(\d{3})(?=\d)/g, "$1" + t) + (c ? d + Math.abs(n - i).toFixed(c).slice(2) : "");
+    }
+
+    static formatTez(a){
+        return this.formatMoney(a)+" ꜩ";
+    }
+
+    static findKeyObj(list, t){
+        for (var i = 0; i < list.length; i++){
+        if (list[i].pkh == t || list[i].label == t) return list[i];
```
Comment (jvtezsure) [inline helper.js]: indentation  w.r.t to opening and closing brackets. Single line IF functions should be avoided. use {} always.

**Review by jvtezsure — CHANGES_REQUESTED** [review]

File: modules/contract/contracts.js [hunk 19]
```diff
@@ -0,0 +1,246 @@
...
+
+  addContractToConfig(contractLabel, contractAddr) {
+      let contractObj = Helper.findKeyObj(config.contracts, contractLabel);
+      if (contractObj) {
+        return Helper.outputError(`This contract label is already in use. Please use a different one.`);
+      }
+      this.__addContract(contractLabel, contractAddr, '');
+  }
+
+  __addContract(label, opHash, pkh) {
```
Comment (jvtezsure) [inline modules/contract/contracts.js]: Why do we need double underscore for function name everywhere? Please remove it from all the places. Only private variables can have single underscore.

**Review by jvtezsure — CHANGES_REQUESTED** [review]

File: modules/contract/contracts.js (hunk 15 above)
Comment (KapilTR) [reply modules/contract/contracts.js]: Have fix this as now I am passing arguments to the function.

File: helper.js [hunk 20]
```diff
@@ -0,0 +1,42 @@
+const { Variables } = require('./utils/cli-variables');
```
Comment (jvtezsure) [inline helper.js]: Move helper js to a modules folder

File: modules/accounts/accounts.js [hunk 21]
```diff
@@ -0,0 +1,224 @@
+const confFile = __dirname + '/../../config.json';
+const jsonfile = require('jsonfile');
+var eztz = {};
+var config = jsonfile.readFileSync(confFile);
```
Comment (jvtezsure) [inline modules/accounts/accounts.js:4]: It can be a const

File: modules/setup/setup.js [hunk 22]
```diff
@@ -0,0 +1,222 @@
...
+            if (!error) {
+                if (__stdout !== `777 ${__dirname}/config.json`) {
+                    child_process.exec(`sudo chmod -R 777 ${__dirname}/config.json`);
+                }
+            }
+        });
+        this.buildImage();
+    }
+
+    buildImage() {
```
Comment (jvtezsure) [inline modules/setup/setup.js]: method name should not be buildImage, it doesn't build the image.

File: modules/setup/setup.js [hunk 23]
```diff
@@ -0,0 +1,222 @@
...
+            stream.on('end', () => {
+                writeStream.end();
+                writeStream.close();
+                console.log(Helper.output(`log files stored as archieve format at: \n     '/tmp/tezster-logs.tar.gz'`))
+            });
+        });
+    }
+
+    checkPermission(){
+        child_process.exec(`stat -c '%a %n' ${__dirname}/config.json`,(error, __stdout, __stderr) => {
```
Comment (jvtezsure) [inline modules/setup/setup.js]: remove all __ from variable names.

File: modules/setup/setup.js [hunk 24]
```diff
@@ -0,0 +1,222 @@
...
+                    progressbar.update(progress);
+                    }, 1000);
+                    docker.modem.followProgress(dockerPullStream, (__dockerModemError, __dockerModemOutput) => {
+                        clearInterval(progress);
+                        progressbar.update(100);
+                        console.log(Helper.output('\nTezos nodes have been setup successfully on system....'));
+                        process.exit();
+                    });
+                }
+            });
```
Comment (jvtezsure) [inline modules/setup/setup.js]: Fix indentation of closing brackets.

File: modules/setup/setup.js [hunk 25]
```diff
@@ -0,0 +1,222 @@
...
+    confirmNodeStatus(){   
+        setTimeout(() => {
+            const _cliProgress = require('cli-progress');
+            const progressbar = new _cliProgress.Bar({
+                format: 'Progress [{bar}] {percentage}%'
+            },
+                _cliProgress.Presets.shades_classic
+            );
+
+            const myInterval = setInterval(() =>{
```
Comment (jvtezsure) [inline modules/setup/setup.js]: variable name should not be  'myInterval', 'yourinterval', 'neighboursInterval' etc.

File: tezster.js [hunk 26]
```diff
@@ -1,467 +1,204 @@
...
-        } else {
-          console.log(
-            tezsterManager.outputError(
-              "Docker not detected on the system please install docker...."
-            )
-          );
-        }
-    });
-  });
+    .version('0.2.0', '-v, --version')
```
Comment (jvtezsure) [inline tezster.js]: version is 0.2.1 on dev.

File: tezster.js [hunk 27]
```diff
@@ -1,467 +1,204 @@
...
+    .command('setup')
+    .description('To set up Tezos nodes')
+    .action(function(){  
+        tezstermanager.setUp();
```
Comment (jvtezsure) [inline tezster.js]: method name should be setup and not setUp. It's a single word.

File: tezster.js [hunk 28]
```diff
@@ -1,467 +1,204 @@
...
-    console.log(result);
+    .command('get-storage')
+    .usage('<contract-name/address>')
+    .description('Returns current storage for given smart contract')
+    .action(function(){
+        tezstermanager.getStorage();
 });
 
-/* Restores an testnet faucet account */
+//******* To Create an contract */
```
Comment (jvtezsure) [inline tezster.js]: why do you need extra '/' before block comment (/* ... */)? You can remove that from everywhere.

**Review by jvtezsure — CHANGES_REQUESTED** [review]

File: modules/setup/setup.js (hunk 23 above)
Comment (KapilTR) [reply modules/setup/setup.js]: __dirname is the directory position of the config.json. From any other variables name I have removed double underscore, using single underscore for private variables.

File: modules/accounts/accounts.js (hunk 21 above)
Comment (KapilTR) [reply modules/accounts/accounts.js:4]: these variables are not being constant throughout the code.

Comment (jvtezsure) [reply modules/accounts/accounts.js:4]: config is constant I think. I'm not asking for all of them, whichever are constant, make it const.

**Review by jvtezsure — CHANGES_REQUESTED** [review]

**Pull request merged**